packages/babel/types/index.d.ts

```diff
@@ -53,10 +53,12 @@ export type RollupBabelCustomOutputPluginOptions = (
   pluginOptions: RollupBabelOutputPluginOptions;
 };
 export type RollupBabelCustomPluginConfig = (
+  this: PluginContext | TransformPluginContext,
```

Comment:
I don't think this should be a union - this only gets called in a single place in the code and thus the type should be always the same. Since it's called from within a `transform` hook I would suspect that this can be simplified to:
```suggestion
  this: TransformPluginContext,
```

Reply:
I did this because the `createBabelOutputPluginFactory` calls the `transformCode` function in the `renderChunk` function where rollup passes only the `PluginContext`. 
Only the `createBabelInputPluginFactory` calls `transformCode` in the `transform` function where rollup passes the `TransformPluginContext`.

Reply:
Good catch. It seems though that we should not reuse the `RollupBabelCustomPluginResult` and `RollupBabelCustomPluginConfig` types for both `RollupBabelCustomInputPlugin` and `RollupBabelCustomOutputPlugin`. I think we should split those so the `this` could be typed correctly for both cases

Reply:
@Andarist Changed things up a bit in: 3b0e78b Please take a look